### PR TITLE
refactor core to create consensus module

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
The `core` crate can be split up to facilitate taking dependencies on modules without pulling in all of `core` which can lead to circular dependencies. Consensus modules can be grouped together to facilitate an eventual separation into a new crate.

#### Summary of Changes
Group consensus modules under `core/src/consensus`.

Fixes #

### CODE REVIEW

File: core/src/tvu.rs [hunk 1]
```diff
@@ -369,7 +369,10 @@ impl Tvu {
 pub mod tests {
     use {
         super::*,
-        crate::validator::TURBINE_QUIC_CONNECTION_POOL_SIZE,
+        crate::{
+            consensus::tower_storage::FileTowerStorage,
```
Comment (CriesofCarrots) [inline core/src/tvu.rs:373]: Isn't this already pulled in with `super::*`?

**Review by CriesofCarrots — APPROVED** [review]
One unimportant nit, but works for me. Would probably be good to wait on @AshwinSekar to check the organization

**Review by AshwinSekar — DISMISSED** [review]
LGTM. Depending on where you're drawing the line you might want to also include some of these:

* `ancestor_hashes_service`: duplicate block consensus
* `cluster_info_vote_listener`: fwd votes for banking stage and optimistic conf
* `cluster_slot_state_verifier`: duplicate block consensus
* `duplicate_repair_status`: duplicate block consensus 
* `optimistic_confirmation_verifier`: optimistic confirmation logic
* `repair_weight`: bank forks++, pruned duplicate block consensus
* `voting_service`: Sends votes to tpu + gossip

File: core/src/tvu.rs (hunk 1 above)
Comment (jbiseda) [reply core/src/tvu.rs:373]: No, only `TowerStorage` is pulled in with `super::*`.

**Comment by jbiseda** [comment]
> LGTM. Depending on where you're drawing the line you might want to also include some of these:
> 
> * `ancestor_hashes_service`: duplicate block consensus
> * `cluster_info_vote_listener`: fwd votes for banking stage and optimistic conf
> * `cluster_slot_state_verifier`: duplicate block consensus
> * `duplicate_repair_status`: duplicate block consensus
> * `optimistic_confirmation_verifier`: optimistic confirmation logic
> * `repair_weight`: bank forks++, pruned duplicate block consensus
> * `voting_service`: Sends votes to tpu + gossip

I'm planning a subsequent PR for repair modules which will pull in some of these files. I think I'll leave this grouping of modules for consensus as-is for now. We can see if anything should be reshuffled prior to creating separate crates.

**Review by AshwinSekar — APPROVED** [review]

File: core/src/tvu.rs (hunk 1 above)
Comment (CriesofCarrots) [reply core/src/tvu.rs:373]: Oh! I'm sorry, my mistake, missed the `File` on this one

**Pull request merged**